src/overlays/actors/ovl_Arrow_Fire/z_arrow_fire.c

```diff
@@ -78,17 +80,18 @@ void ArrowFire_Charge(ArrowFire* this, GlobalContext* globalCtx) {
 
     // if arrow has no parent, player has fired the arrow
     if (arrow->actor.parent == NULL) {
-        this->unkPos = this->actor.world.pos;
+        this->trailEnd = this->actor.world.pos;
         this->radius = 10;
         ArrowFire_SetupAction(this, ArrowFire_Fly);
         this->alpha = 255;
     }
 }
 
-void func_80865ECC(Vec3f* unkPos, Vec3f* firePos, f32 scale) {
-    unkPos->x += ((firePos->x - unkPos->x) * scale);
-    unkPos->y += ((firePos->y - unkPos->y) * scale);
-    unkPos->z += ((firePos->z - unkPos->z) * scale);
+void shrinkFireTrail(Vec3f* trailEnd, Vec3f* currentPos, f32 scale) {
+    // shrinks trail by 5% (scale) of the difference between the current position and trail end
```

Comment:
I don't think the comment should tell the typical value of `scale` since that comes from outside the scope of the function
(if you keep modding in mind, functions may be reused for whatever purpose)

I also feel like (may just be me though) that the function in general should be named after what it does (move vector A towards vector B by a fraction of the AB distance), instead of what it is used for
`Math_ApproachF` looks similar apart from the `step` parameter so you may want to reuse "Approach" here